glommio/src/executor.rs

```diff
@@ -559,6 +559,276 @@ impl Default for LocalExecutorBuilder {
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// A factory that allows creating a pool of [`LocalExecutor`]s, which can be
+/// used in order to configure the properties of the [`LocalExecutor`]s.
+/// Configuration methods apply their settings to all [`LocalExecutor`]s in the
+/// pool unless otherwise specified.
+///
+/// Methods can be chained on the builder in order to configure it.
+///
+/// The [`LocalExecutorPoolBuilder::on_all_shards`] method will take ownership
+/// of the builder and create a [`PoolThreadHandles`] struct which can be used
+/// to join the executor threads.
+// TODO: decide whether to avoid redundancy with with `LocalExecutorBuilder` by
+// using a shared inner type or a generic type mix-in with a type definition
+#[derive(Debug)]
+pub struct LocalExecutorPoolBuilder {
+    /// The number of [`LocalExecutor`]s the builder should attempt to create.
+    nr_shards: usize,
+    // TODO: `spin_before_park` will become operational when this builder can actually bind to a
+    // cpu
+    // /// Spin for duration before parking a reactor
+    // spin_before_park: Option<Duration>,
+    /// A name for the thread-to-be (if any), for identification in panic
+    /// messages; each executor in the pool will use this name followed by
+    /// a hyphen and numeric notifier id (e.g. `myname_0`)
+    name: String,
+    /// Amount of memory to reserve for storage I/O. This will be preallocated
+    /// and registered with io_uring. It is still possible to use more than
+    /// that but it will come from the standard allocator and performance
+    /// will suffer. Defaults to 10MB.
+    io_memory: usize,
+    /// How often to yield to other task queues
+    preempt_timer_duration: Duration,
+    /// Indicates a policy by which [`LocalExecutor`]s are bound to CPUs.
+    placement: Placement,
+}
+
+impl LocalExecutorPoolBuilder {
+    /// Generates the base configuration for spawning a pool of
+    /// [`LocalExecutor`]s, from which configuration methods can be chained.
+    /// The method's only argument sets the number of [`LocalExecutor`]s to
+    /// spawn.
+    pub fn new(nr_shards: usize) -> Self {
+        Self {
+            nr_shards,
+            // spin_before_park: None,
+            name: String::from("unnamed"),
+            io_memory: 10 << 20,
+            preempt_timer_duration: Duration::from_millis(100),
+            placement: Placement::Unbound,
+        }
+    }
+
+    // /// Please see documentation under
+    // /// [`LocalExecutorBuilder::spin_before_park`] for details.  The setting
+    // /// is applied to all executors in the pool.
+    // pub fn spin_before_park(mut self, spin: Duration) -> Self {
+    //     self.spin_before_park = Some(spin);
+    //     self
+    // }
+
+    /// Please see documentation under [`LocalExecutorBuilder::name`] for
+    /// details. The setting is applied to all executors in the pool.
+    pub fn name(mut self, name: &str) -> Self {
```

Comment:
that's usually not what the user wants. Usually we want each thread to have a unique name, so for an executor pool this is likely a function, taking an `id` as a parameter.

Reply:
This comment is actually somewhat misleading anyway.  As with the `LocalExecutorBuilder`, the name will ultimately be combined with a notifier id, so it would be something like `name-5`.  I'm happy to modify this--maybe storing a `Box<dyn Fn(usize) -> String>` in the builder (I'm not aware of an ergonomic way to use a closure as an optional field with static dispatch).

Reply:
"name-x" is okay.
No need for a function in this case. This is a convenience interface, so there is no need to go for ultra flexibility.

Thanks!